spec/googleauth/get_application_default_spec.rb

```diff
@@ -125,17 +125,35 @@
       end
     end
 
-    it "succeeds without default file or env if on compute engine" do
-      stub = stub_request(:get, "http://169.254.169.254")
-             .to_return(status:  200,
-                        headers: { "Metadata-Flavor" => "Google" })
-      Dir.mktmpdir do |dir|
-        ENV.delete @var_name unless ENV[@var_name].nil? # no env var
-        ENV["HOME"] = dir # no config present in this tmp dir
-        creds = Google::Auth.get_application_default @scope, options
-        expect(creds).to_not be_nil
+    describe "when on compute engine" do
+      before do
+        @compute_metadata_server = stub_request(:get, "http://169.254.169.254")
```

Comment:
@bajajneha27 thanks for feedback.
 While one would _normally_ use a `let!(:compute_metadata_server)` as it's more idiomatic to RSpec test suite, this file seemed to be only be using `@instance_variable` so for the sake of consistency I've kept the pattern used in the file.

There seem to be a mixed use of `let` and `before { @instance_varialbe = "foo" }` throughout the test suite. It's also unclear what is the preferred way.
As a long term rubyist I'd be inclined to favour the `let {}`.
I'd be happy to convert this file from using instance  variable to `let` blocks, in this pull request or another one.

Reply:
This looks fine. Thanks for implementing this.